sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_dsl_pipeline_with_specific_nodes.py

```diff
@@ -49,6 +70,89 @@
 @pytest.mark.e2etest
 @pytest.mark.pipeline_test
 class TestDSLPipelineWithSpecificNodes(AzureRecordedTestCase):
+    @staticmethod
+    def _generate_multi_layer_pipeline_func():
+        path = "./tests/test_configs/components/helloworld_component.yml"
+
+        @dsl.pipeline
+        def pipeline_leaf(component_in_path: Input):
+            component_func1 = load_component(source=path)
+            component_func1(component_in_path=component_in_path, component_in_number=1)
+
+            component_func2 = load_component(source=path, params_override=[{
+                "name": "another_component_name",
+                "version": "another_component_version",
+            }])
+            component_func2(component_in_path=component_in_path, component_in_number=1)
+
+            component_func3 = load_component(source=path, params_override=[{
+                "environment": "azureml:AzureML-sklearn-0.24-ubuntu18.04-py37-cpu:2"
+            }])
+            component_func3(component_in_path=component_in_path, component_in_number=1)
+
+            component_func4 = load_component(source=path)
+            component_func4.command += " & echo updated1"
+            component_func4(component_in_path=component_in_path, component_in_number=1)
+
+        @dsl.pipeline
+        def pipeline_mid(job_in_path: Input):
+            pipeline_leaf(job_in_path)
+            pipeline_leaf(job_in_path)
+
+        @dsl.pipeline
+        def pipeline_root(job_in_path: Input):
+            pipeline_mid(job_in_path)
+            pipeline_mid(job_in_path)
+        return pipeline_root
+
+    @staticmethod
+    def _generate_pipeline_func_for_concurrent_component_registration_test(shared_input):
+        path = "./tests/test_configs/components/helloworld_component.yml"
+        conda_file_path = "./tests/test_configs/environment/environment_files/environment.yml"
+
+        environment = Environment(
+            name="test-environment",
+            conda_file=conda_file_path,
+            image="mcr.microsoft.com/azureml/openmpi3.1.2-ubuntu18.04",
```

Comment:
may need to update the image to ubuntu 20

Reply:
updated